modal/app.py

```diff
@@ -473,7 +474,7 @@ def wrapped(raw_f: Callable[..., Any]) -> None:
 
     def function(
         self,
-        _warn_parentheses_missing=None,
+        _warn_parentheses_missing: bool | None = None,
```

Comment:
This needs to be spelled as `Optional[bool]` as we support Python <3.10

Reply:
We can add `from __future__ import annotations` and the new syntax will be supported if we want, but let me switch this out, bad habit on my end.

Reply:
https://github.com/modal-labs/modal-client/pull/1875/commits/1a0dadd7cbbaa4e9ab986708f3feb546870a6a9e